R/mail.R

```diff
@@ -6,110 +6,249 @@
 
 ############## Downloader ##############
 
-#' Download all pipermail files in an archive
-#' @param url An url pointing to a pipermail archive
-#' @return Returns `destination`, a vector of the downloaded files in the current working directory
+#' Download all pipermail files in an archive as mbox files
+#' @param mailing_list The name of the mailing list being downloaded
+#' @param start_year_month The year and month of the first file to be downloaded
+#' @param end_year_month The year and month of the last file to be downloaded
+#' @param save_folder_path The folder path in which all the downloaded pipermail files will be stored
+#' @return Returns `downloaded_files`, a vector of the downloaded files in the current working directory
 #' @export
-download_pipermail <- function(url) {
+download_pipermail <- function(mailing_list, start_year_month, end_year_month, save_folder_path) {
 
-  #Get page
-  pagedata <- httr::GET(url)
-
-  #Parse html file into object
-  tbls_xml <- XML::htmlParse(pagedata)
-
-  #Extract href tablenodes from html table
-  tableNodes <- XML::getNodeSet(tbls_xml, "//td/a[@href]")
+  # Create directory if it does not exist
+  if (!dir.exists(save_folder_path)) {
+    dir.create(save_folder_path, recursive = TRUE)
+  }
 
-  #Extract filenames from tablenode content with xmlGetAtrr
-  hrefs <- sapply(tableNodes, XML::xmlGetAttr, 'href')
+  # Ensure mailing_list URL ends with a slash
+  if (!stringi::stri_endswith_fixed(mailing_list, "/")) {
+    mailing_list <- paste0(mailing_list, "/")
+  }
 
-  #Create Vector
-  files <- vector()
+  # Get mailing list contents
```

Comment:
I think this is the place where the file with all the mail urls are displayed? I would be more verbose here, since that starts the first part of the code logic to scan the file.